From 98ada1a06739dd960e9d27f947cef373fa4b3afc Mon Sep 17 00:00:00 2001
From: Robert Kanter <rkanter@apache.org>
Date: Thu, 9 Apr 2015 13:48:14 -0700
Subject: [PATCH 0527/1023] MAPREDUCE-6266. Job#getTrackingURL should
 consistently return a proper URL (rchiang via
 rkanter)

(cherry picked from commit c4986b2d00d327f18d0c7e2f8805b69a4e07a19b)

Conflicts:
	hadoop-mapreduce-project/CHANGES.txt
---
 .../hadoop/mapreduce/v2/hs/CompletedJob.java       |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs/src/main/java/org/apache/hadoop/mapreduce/v2/hs/CompletedJob.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs/src/main/java/org/apache/hadoop/mapreduce/v2/hs/CompletedJob.java
index 049a389..8a7bf00 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs/src/main/java/org/apache/hadoop/mapreduce/v2/hs/CompletedJob.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs/src/main/java/org/apache/hadoop/mapreduce/v2/hs/CompletedJob.java
@@ -144,10 +144,10 @@ private void constructJobReport() {
     String historyUrl = "N/A";
     try {
       historyUrl =
-          MRWebAppUtil.getApplicationWebURLOnJHSWithoutScheme(conf,
+          MRWebAppUtil.getApplicationWebURLOnJHSWithScheme(conf,
               jobId.getAppId());
     } catch (UnknownHostException e) {
-      //Ignore.
+        LOG.error("Problem determining local host: " + e.getMessage());
     }
     report.setTrackingUrl(historyUrl);
     report.setAMInfos(getAMInfos());
-- 
1.7.9.5

